### ISSUE TITLE
Proposal: Interfaces should allow duplicated members 

### ISSUE THREAD

**Issue by ENikS** [Microsoft/TypeScript#4423]
Allow duplicated members in multiple interface declarations, as long as they are identical. Optional compiling warnings are acceptable, to assist developers to remove redundant codes at convenient time.

This should not generate any errors, example:

shim.d.ts
```javascript
interface Symbol {
    toString(): string;
    valueOf(): Object;
    [Symbol.toStringTag]: string;
}
```

es6.d.ts
```javascript
...
interface Symbol {
    /** Returns a string representation of an object. */
    toString(): string;

    /** Returns the primitive value of the specified object. */
    valueOf(): Object;

    [Symbol.toStringTag]: string;
}
...
```

**Comment by mhegazy** [comment]
I do not think we need a new issue for this. issue #3215 seems like a fine place to keep the discussion going.
